services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -13,38 +13,63 @@ import scala.concurrent.{ ExecutionContext, Future }
 class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
     extends ProjectVersioning[VersioningException] {
 
-  override def updateFile(project: Project, projectFile: ProjectFile, version: Option[Version])(
+  override def updateFile(project: Project, projectFile: ProjectFile, version: Option[PipelineVersion])(
     implicit ec: ExecutionContext
   ): AsyncResult[String] = {
     val path = URLEncoder.encode(projectFile.path.toString, "UTF-8")
-    val fileUrl = s"${config.url}projects/${project.repository}/repository/files/$path"
-    val versionValue = version.map(_.name).getOrElse(config.defaultFileVersion)
+    val repositoryId: Repository = project.repository match {
+      case Some(repository) => repository
+      case None             => throw VersioningException(s"No repository for project: $project")
+    }
+    val fileUrl = s"${config.url}projects/${repositoryId.value}/repository/files/$path"
 
+    getProjectVersions(project).flatMap(versions => {
+      val latestVersion: Either[VersioningException, PipelineVersion] = (version, versions) match {
```

Comment:
Why we're taking into account new version, coming from user, if the variable is `latestVersion`?

Reply:
Didn't catch your question - latestVersion here is really latest version between project versions and user version.